modules/chat/tests/server/chat.socket.tests.js

```diff
@@ -3,6 +3,117 @@
 /**
  * Chat socket tests
  */
-describe('Chat Socket Tests:', function () {
-  // TODO: Add chat socket tests
+
+var should = require('should'),
+  request = require('supertest'),
+  path = require('path'),
+  mongoose = require('mongoose'),
+  sinon = require('sinon'),
+  User = mongoose.model('User'),
+  express = require(path.resolve('./config/lib/express')),
+  io = require('socket.io/node_modules/socket.io-client');
+
+/**
+ * Globals
+ */
+var app, agent, credentials, ioc, user, admin, sessionId, clock;
+
+describe('Socket Chat tests:', function() {
+
+  // Get application
+  before(function(done) {
+    app = express.init(mongoose);
+    agent = request.agent(app);
+
+    done();
+  });
+
+  // create the user
+  beforeEach(function(done) {
+    // Create user credentials
+    credentials = {
+      username: 'username',
+      password: 'password'
+    };
+
+    // Create a new user
+    user = new User({
+      firstName: 'Full',
+      lastName: 'Name',
+      displayName: 'Full Name',
+      email: 'test@test.com',
+      username: credentials.username,
+      password: credentials.password,
+      provider: 'local'
+    });
+
+    // Save a user to the test db
+    user.save(function () {
+      done();
+    });
+  });
+
+  it('should be able to connect on socket', function(done) {
+    clock = sinon.useFakeTimers(Date.now());
```

Comment:
Added `sinon.useFakeTimers` as we should use it anyway when testing. In 1 or 10 cases my tests fail because of some latency. Maybe we can shift to this when testing objects where time is just on of the attributes.